internal/configs/version1/nginx-plus.ingress.tmpl

```diff
@@ -1,7 +1,7 @@
 # configuration for {{.Ingress.Namespace}}/{{.Ingress.Name}}
 {{range $upstream := .Upstreams}}
 upstream {{$upstream.Name}} {
-	zone {{$upstream.Name}} 256k;
+	zone {{$upstream.Name}} {{$upstream.UpstreamZoneSize}};
```

Comment:
I suggest ignoring the special value `0` - in that case we can use the default. This would be in contrast with NGINX, for which `0` disables the zone.